ra/ra.go

```diff
@@ -972,6 +972,10 @@ func (ra *RegistrationAuthorityImpl) failOrder(
 // If successful the order will be returned in processing status for the client
 // to poll while awaiting finalization to occur.
 func (ra *RegistrationAuthorityImpl) FinalizeOrder(ctx context.Context, req *rapb.FinalizeOrderRequest) (*corepb.Order, error) {
+	if req == nil || req.Order == nil {
+		return nil, berrors.InternalServerError(errIncompleteGRPCRequest.Error())
```

Comment:
Everywhere else we return `errIncompleteGRPCRequest` we have not wrapped it in `berrors.InternalServerError()`. IMO we should return `errIncompleteGRPCRequest` by itself here and either in this change or a follow up change set `var errIncompleteGRPCRequest = berrors.InternalServerError("incomplete gRPC request message")`.

Reply:
Yeah, good catch here, Andrew. We don't want to return `berrors.InternalServerError` here. This is because, when `web/probs.go` converts an internal error into a user-facing error, it does the following things:
1) If the error is already a `berror`, then it converts it to the corresponding `Problem` and preserves the message
2) otherwise, it assumes the error is "unhandled", and converts it into a `probs.ServerInternal` and _does not_ preserve the message (because who knows what internal info is in that message).

In this case, we don't want to preserve the message and expose it to the user. So we want `problemDetailsForError` to throw away the message and just replace it with a generic "internal server error". So we need to make sure that this error is not already a `berror`.